### PR TITLE
Remove await* from babel-generator, add parsing error to babylon - (fixes T6688)

### PR DESCRIPTION
**Pull request by hzoo**
https://phabricator.babeljs.io/T6688

I think this works?

Ok need to fix up some tests

### CODE REVIEW

File: packages/babylon/src/parser/expression.js [hunk 1]
```diff
@@ -968,7 +968,10 @@ pp.parseAwait = function (node) {
   if (this.isLineTerminator()) {
     this.unexpected();
   }
-  node.all = this.eat(tt.star);
+  node.all = false;
```
Comment (hzoo) [inline packages/babylon/src/parser/expression.js]: Can remove this and then fixup the tests (since otherwise it's node.all being undefined)

Comment (amasad) [inline packages/babylon/src/parser/expression.js]: yeah, I don't see any use for it.

**Comment by amasad** [comment]
LGTM, just remove the node.all

**Comment by hzoo** [comment]
Ok removed and fixed tests

**Pull request merged**